src/c++/perf_analyzer/genai-perf/genai_perf/export_data/json_exporter.py

```diff
@@ -0,0 +1,71 @@
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+import json
+from enum import Enum
+from pathlib import Path
+from typing import Dict
+
+from genai_perf.constants import DEFAULT_OUTPUT_DATA_JSON
+from genai_perf.export_data.file_exporter import FileExporter
+
+
+class JsonExporter(FileExporter):
+    """
+    A file exporter class that adheres to the FileExporter protocol
+    """
+
+    def __init__(self, stats: Dict, args: Dict, extra_inputs: Dict):
+        self._stats = stats
+        self._args = dict(args)
+        self._extra_inputs = extra_inputs
+        self._stats_and_args: Dict = {}
+        self._prepare_args_for_export()
+        self._merge_stats_and_args()
+
+    def export_to_file(self, output_dir: Path) -> None:
+        filename = output_dir / DEFAULT_OUTPUT_DATA_JSON
+        with open(str(filename), "w") as f:
+            f.write(json.dumps(self._stats_and_args, indent=2))
+
+    def _prepare_args_for_export(self) -> None:
+        del self._args["func"]
+        del self._args["output_format"]
+        self._args["profile_export_file"] = str(self._args["profile_export_file"])
+        self._args["artifact_dir"] = str(self._args["artifact_dir"])
+        for k, v in self._args.items():
+            if isinstance(v, Enum):
+                self._args[k] = v.name.lower()
+        self._add_extra_inputs_to_args()
+
+    def _add_extra_inputs_to_args(self) -> None:
+        del self._args["extra_inputs"]
+        self._args.update({"extra_inputs": self._extra_inputs})
```

Comment:
Why delete and set again?

Reply:
I am converting the extra_inputs to a dict. Currently it is stored as a list.